miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/userinfo/UserInfoBridgeDispatcher.kt

```diff
@@ -22,16 +22,44 @@ abstract class UserInfoBridgeDispatcher {
      * Get user name from host app.
      * You can also throw an [Exception] from this method to pass an error message to the mini app.
      */
+    @Deprecated(
+        "This function has been deprecated.",
+        ReplaceWith("getUserName(onSuccess: (name: String) -> Unit, onError: (message: String) -> Unit)")
+    )
     open fun getUserName(): String =
         throw MiniAppSdkException("The `UserInfoBridgeDispatcher.getUserName` $NO_IMPL")
 
+    /**
+     * Get user name from host app.
+     * You can send user name or throw an [Exception] from this method by passing a [Result].
+     */
+    open fun getUserName(
+        callback: (result: Result<String?>) -> Unit
+    ) {
+        throw MiniAppSdkException("The `UserInfoBridgeDispatcher.getUserName` $NO_IMPL")
+    }
+
     /**
      * Get profile photo url from host app.
      * You can also throw an [Exception] from this method to pass an error message to the mini app.
      */
+    @Deprecated(
+        "This function has been deprecated.",
+        ReplaceWith("getProfilePhoto(onSuccess: (photoUrl: String) -> Unit, onError: (message: String) -> Unit)")
+    )
     open fun getProfilePhoto(): String =
         throw MiniAppSdkException("The `UserInfoBridgeDispatcher.getProfilePhoto` $NO_IMPL")
 
+    /**
+     * Get profile photo url from host app.
+     * You can send profile photo url or throw an [Exception] from this method by passing a [Result].
+     */
+    open fun getProfilePhoto(
+        callback: (result: Result<String?>) -> Unit
```

Comment:
Yeah, we can do that. But it seems like we are intending to add many deprecated functions at a once. I think, if we write this asynchronous username and profile photo function without using `Result` to keep consistency with others, then it will be user friendly. 
Then, we can add `Result` in a separate ticket, of course it will be nice if we can completely remove the deprecated functions meanwhile. 

@corycaywood @minh-rakuten What do you think?

Reply:
> Then, we can add Result in a separate ticket, of course it will be nice if we can completely remove the deprecated functions meanwhile.

Or, I can address it in this same PR as @corycaywood suggested before. So, I have updated this PR.

I want to address `shareContent` with using `Result` in a separate PR because it has default implementation. So, need to think differently.